specs/agents/metrics-health.md

```diff
@@ -0,0 +1,132 @@
+## Agent Health Metrics
+
+Agents SHOULD expose metrics which allow the analysis of the internal state of the agent, especially including the [reporting/transport mechanism](transport.md).
+
+Many of these metrics are `COUNTER`s: a `COUNTER` is a monotonically increasing metric.
+It is periodically reported by sending the difference of the counter-value now vs its value at the time of the previous report.
+In order to save elasticsearch disk space, zero-increases are not sent.
+
+Example:
+| **Time** | **Counter** | **Reported Metric Value** |
+|----------|-------------|---------------------------|
+| 0s       | 7           | 7                         |
+| 30s      | 10          | 3                         |
+| 1m       | 10          | 0 (not sent)              |
+| 1m30s    | 12          | 2                         |
+
+
+### Event Count Metrics
+
+Agents send their data to the Intake API of the APM-server in the form of a stream of events (e.g. transactions, spans, metricsets...).
+
+The following metrics counting these events SHOULD be exposed:
+ - `agent.events.total`: `COUNTER` of events either submitted to the internal reporting queue or dropped upon submit attempt
+ - `agent.events.dropped.queue`: `COUNTER` of events where the attempt to add it to the reporting queue failed (e.g. due to a full queue)
+ - `agent.events.dropped.error`: `COUNTER` of events dropped after they previously have been successfully added to the queue.
```

Comment:
Have you considered just having a `agent.events.dropped` metrics with a `reason` label?

Reply:
I used the metrics suggested in your space-time project as starting point ([document](https://docs.google.com/document/d/1Bi-Km-HRREWOtahgYyvVzQ5LzTD50Wfq-9feeTDExDU/edit#heading=h.7fsov1jn8ngv)).

I don't see a benefit / downside on either approach, so I'm free to change it if you have a preference.

Reply:
If there's nothing that speaks against it, a label would be my preference. This also allows easier extension in the future, for example, differentiating different kinds of `error` scenarios, such as validation errors vs network errors.